lib/net/winrm/connection.rb

```diff
@@ -0,0 +1,289 @@
+              # http://msdn.microsoft.com/en-us/library/cc251676.aspx
+              if err.fault_code == '2150858793'
+                yield nil, nil
+              else
+                raise
+              end
+            end
+          end
+    
+          def send_stdin(input)
+            open unless shell_id
+            create_proc unless command_id
+    
+            stdin_msg = WinRM::WSMV::WriteStdin.new(
+              connection_opts,
+              shell_uri: shell_uri,
+              shell_id: shell_id,
+              command_id: command_id,
+              stdin: input,
+            )
+            result = transport.send_request(stdin_msg.build)
+            result
+          rescue WinRM::WinRMWSManFault => e
+            raise unless [ERROR_OPERATION_ABORTED, SHELL_NOT_FOUND].include?(e.fault_code)
+          rescue WinRM::WinRMHTTPTransportError => t
+            # dont let the cleanup raise so we dont lose any errors from the command
+            logger.info("[WinRM] #{t.status_code} returned in cleanup with error: #{t.message}")
+          end
+    
+    
+          def response_reader
+            @response_reader ||= ReceiveResponseReader.new(transport, logger)
+          end
+    
+          def open_shell
+            msg = WinRM::WSMV::CreateShell.new(connection_opts, shell_opts)
+            resp_doc = transport.send_request(msg.build)
+            self.owner = REXML::XPath.first(resp_doc, "//rsp:Owner").text
+            REXML::XPath.first(resp_doc, "//*[@Name='ShellId']").text
+          end
+    
+          attr_accessor :owner
+    
+          protected
+            attr_accessor :command_id
+        end
+    
+        def create_shell(shell_type, shell_opts = {})
+          args = [
+            @connection_opts,
+            @transport,
+            @logger
+          ]
+          return StdinShell.new(*args) if shell_type == :stdin
+          super(shell_type, shell_opts)
+        end
+      end
+    
+      class TransportFactory < WinRM::HTTP::TransportFactory
+        class RexHttpTransport < WinRM::HTTP::HttpTransport
+          # rubocop:disable Lint/
+          def initialize(opts)
+            self.http_client = Rex::Proto::Http::Client.new(opts[:host], opts[:port], {}, opts[:ssl], opts[:ssl_version], opts[:proxies], opts[:user], opts[:password])
+            self.timeout = opts[:timeout]
+            @mutex = Mutex.new
+            self.uri = opts[:uri]
+            if opts[:realm]
+              self.http_client.set_config('domain' => opts[:realm])
+            end
+          end
+
+          def peerinfo
+            if self.http_client and self.http_client.conn
+              self.http_client.conn.peerinfo
+            end
+          end
+
+          def localinfo
+            if self.http_client and self.http_client.conn
+                self.http_client.conn.localinfo
+            end
+          end
+
+          def ntlm_transform_response(ntlm_client, response)
```

Comment:
Just to ask the obvious, what are the reasons for overriding these methods? 👀

Reply:
Good call; comments added. As far as I can tell from the content-type, I can't find anything else anywhere that uses this, other than WSMV, so didn't want to put it directly into the HttpClient itself.